### PR TITLE
chore(docs): adding aws step functions recipe documentation

### PR DESCRIPTION
**Pull request by xoscar**
This PR updates the documentation website by adding the AWS Step Functions recipe page

## Fixes

- #2084 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+[AWS Fargate](https://aws.amazon.com/fargate/) is a serverless, pay-as-you-go compute engine that lets you focus on building applications without managing servers. AWS Fargate is compatible with both Amazon Elastic Container Service (ECS) and Amazon Elastic Kubernetes Service (EKS).
+[AWS Step Functions](https://aws.amazon.com/step-functions) is a visual workflow service that helps developers use AWS services to build distributed applications, automate processes, orchestrate microservices, and create data and machine learning (ML) pipelines.
+[AWS X-Ray](https://aws.amazon.com/xray/) provides a complete view of requests as they travel through your application and filters visual data across payloads, functions, traces, services, APIs, and more with no-code and low-code motions.
+
+## .NET Step Functions Serverless API with AWS X-Ray, AWS Fargate and Tracetest
+
+The use case you'll see in this recipe is based on [this .NET demo repository](https://github.com/aws-samples/aws-step-functions-plagiarism-demo-dotnetcore) that illustrates how to create a basic State Machine based using Step Functions, Lambda and DynamoDB.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:15]:
```suggestion
The use case you'll see in this recipe is based on [this .NET demo repository](https://github.com/aws-samples/aws-step-functions-plagiarism-demo-dotnetcore) that illustrates how to create a Basic State Machine using Step Functions, Lambda and DynamoDB.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+![AWS Serverless Diagram](../img/aws-step-functions.png)
+
+## Prerequisites
+
+You will need [Terraform](https://www.terraform.io/), a configured instance of the [AWS CLI](https://aws.amazon.com/cli/) and the [SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/install-sam-cli.html) installed on your machine to run this quick start Step Functions app!
+
+## Project Structure
+
+The project is divided into two main sections. The `infra` folder includes the basic services and resources that are required to run Tracetest iin your AWS account using Terraform as deployment framework. Meanwhile, the `src` folder includes the code and configuration to create and provision the Step Functions using the SAM framework.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:29]:
```suggestion
The project is divided into two main sections. The `infra` folder includes the basic services and resources that are required to run Tracetest in your AWS account using Terraform as the deployment framework. Meanwhile, the `src` folder includes the code and configuration to create and provision the Step Functions using the SAM framework.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+The State Machine definition can be found in the `scr/state-machine.asl.json` file which is written using the own [AWS States Language](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-amazon-states-language.html).
+
+### 2. Tracetest
+
+Under the `infra` folder you'll find the `tracetest.tf` file which includes the required services and configuration to run Tracetest on AWS Fargate.
+
+### AWS Network
+
+To connect all of the services, the example generates a VPC network which includes private and public subnets. Internal service like the Postgres RDS instance are protected behind the VPC and only accessible through a node within the internal network.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:43]:
```suggestion
To connect all of the services, the example generates a VPC network which includes private and public subnets. Internal services like the Postgres RDS instance are protected behind the VPC and only accessible through a node within the internal network.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+The `infra/tracetest.tf` file contains the required services for the Tracetest server which include.
+
+- **Postgres RDS** - Postgres is a prerequisite for Tracetest to work. It stores trace data when running the trace-based tests.
+- [**Tracetest Task Definition**](https://tracetest.io/) - The information on how to configure and provision Tracetest using ECS.
+- [**ECS Service**](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs_services.html) - The server provisioning metadata to run the Tracetest Task Definition.
+- **Networking** - Security groups, target groups and load balancer listeners required to have Tracetest connected to the rest of the AWS infrastructure.
+
+### Configuring the Tracetest Container
+
+The Tracetest docker image supports environment variables as entry point for the bootstrap configuration, in this case the task definition incliudes the following:
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:58]:
```suggestion
The Tracetest docker image supports environment variables as the entry point for the bootstrap configuration. In this case the task definition includes the following:
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      "name" : "TRACETEST_POSTGRES_PASSWORD",
+      "value" : "${module.db.db_instance_password}"
+    },
+  ]
+}
+```
+
+## AWS X-Ray
+
+This recipe uses AWS X-Ray as Data Source for the telemetry data, it doesn't require any predefined configuration to start using this service only having the right access for applications to been able to send and fetch information from it.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:90]:
```suggestion
This recipe uses AWS X-Ray as the Data Source for the telemetry data. It doesn't require any predefined configuration to start using this service, only needing the correct access for applications to be able to send and fetch information from it.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+## How do Tracetest reach AWS X-Ray?
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:92]:
```suggestion
## How does Tracetest reach AWS X-Ray?
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+Tracetest supports the native AWS SDK library to connect to X-Ray to find traces. Today the only authentication method it supports is by manually adding a set of AWS credentials that have access to the X-Ray service from your account.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:94]:
```suggestion
Tracetest supports the native AWS SDK library to connect to X-Ray to find traces. Today the only authentication method supported is manually adding a set of AWS credentials that have access to the X-Ray service from your account.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+```
+
+You'll be asked to fill some details like the app name, email settings and a Sendgrid API key.
+The output from the previous command should look similar ot this:
+
+![SAM Step Functions](../img/sam-step-functions.png)
+
+## Running Trace Based Tests
+
+Now that all of the required services and infra has been created, you can start running some Trace based testing by doing the following:
```
Comment (kdhamric) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:150]: Trace-based

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+sam build \
+sam deploy --guided
+```
+
+You'll be asked to fill some details like the app name, email settings and a Sendgrid API key.
+The output from the previous command should look similar ot this:
+
+![SAM Step Functions](../img/sam-step-functions.png)
+
+## Running Trace Based Tests
```
Comment (kdhamric) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:148]: Trace-based

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+    sessionToken: <your-session-token>
+    region: "us-west-2"
+```
+
+Or from the the Tracetest settings page:
+![AWS X-Ray Settings](../img/awsxray-settings.png)
+
+## How do traces reach AWS X-Ray?
+
+This recipe uses the auto-instrumentation [library for .NET](https://docs.aws.amazon.com/xray-sdk-for-dotnet/latest/reference/html/N_Amazon_XRay_Recorder_Handlers_AwsSdk.htm) which automatically listens to any aws sdk request and sends the telemetry data to X-Ray.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:112]:
```suggestion
This recipe uses the auto-instrumentation [library for .NET](https://docs.aws.amazon.com/xray-sdk-for-dotnet/latest/reference/html/N_Amazon_XRay_Recorder_Handlers_AwsSdk.htm) which automatically listens to any AWS SDK request and sends the telemetry data to X-Ray.
```

**Review by kdhamric — APPROVED** [review]

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+
+To provide the lambda functions access to X-Ray the `arn:aws:iam::aws:policy/AWSXrayWriteOnlyAccess` AWS managed policy is also added.
+
+## Running the example
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:116]:
```suggestion
## Running the Example
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+
+The first thing you need to do is to provision the Tracetest infrastructure by running the following commands from the `infra` folder.
+
+```bash
+terraform init \
+terraform apply
+```
+
+After accepting the changes after running the `terraform apply` command and finalizing the infra creation you can find the set of output with the required endpoints to continue with some tests.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:125]:
```suggestion
After accepting the changes after running the `terraform apply` command and finalizing the infra creation, you can find the output with the required endpoints to continue with tests.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+The final output from the Terraform command should be a list of endpoints that similar to the following:
+
+![Terraform Step Functions Output](../img/terraform-step-functions.png)
+
+You can follow the `tracetest_url` to find the Tracetest UI.
+
+Second step is adding the AWS credentials for Tracetest to reach to X-Ray by following this url `tracetest_url/settings`. Selecting the X-Ray as data store and updating the settings.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:132]:
```suggestion
The second step is adding the AWS credentials for Tracetest to reach to X-Ray by following this url `tracetest_url/settings`. Select the X-Ray as data store and updating the settings.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+
+**Tip.** After you add the credentials, you can test the connection by clicking the Test Connection button.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:134]:
```suggestion
**Tip.** After you add the credentials, you can test the connection by clicking the **Test Connection** button.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+
+Third step is to create the Lambda functions and the rest of the services to run the Step Functions, running the following command from the `src` folder:
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:136]:
```suggestion
The third step is to create the Lambda functions and the rest of the services to run the Step Functions by running the following command from the `src` folder:
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+
+```bash
+sam build \
+sam deploy --guided
+```
+
+You'll be asked to fill some details like the app name, email settings and a Sendgrid API key.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:143]:
```suggestion
You'll be asked to fill in some details like the app name, email settings and a Sendgrid API key.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md (hunk 9 above)
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:148]:
```suggestion
## Running Trace-based Tests
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md (hunk 8 above)
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:150]:
```suggestion
Now that all of the required services and infra has been created, you can start running trace-based testing by doing the following:
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+The output from the previous command should look similar ot this:
+
+![SAM Step Functions](../img/sam-step-functions.png)
+
+## Running Trace Based Tests
+
+Now that all of the required services and infra has been created, you can start running some Trace based testing by doing the following:
+
+1. Run this `sam list stack-outputs --stack-name <your_app_name>` sam command where you can copy the `StepFunctionsAPIUrl` value and add replace the `<your_api_endpoint>` placeholder from the `tests/incident.yaml` and `tests/exam.yaml` files
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:152]:
```suggestion
1. Run this `sam list stack-outputs --stack-name <your_app_name>` sam command where you can copy the `StepFunctionsAPIUrl` value and add the `<your_api_endpoint>` placeholder from the `tests/incident.yaml` and `tests/exam.yaml` files.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 18]
```diff
@@ -0,0 +1,166 @@
...
+2. From the Terraform output configure the [Tracetest CLI](https://docs.tracetest.io/cli/configuring-your-cli) to point to the public load balancer endpoint with `tracetest configure --endpoint <tracetest_url>`
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:153]:
```suggestion
2. From the Terraform output, configure the [Tracetest CLI](https://docs.tracetest.io/cli/configuring-your-cli) to point to the public load balancer endpoint with `tracetest configure --endpoint <tracetest_url>`.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 19]
```diff
@@ -0,0 +1,166 @@
...
+3. Run the tests YAML file using the CLI
+
+```bash
+  tracetest test run -d tests/incident.yaml \
+  tracetest test run -d tests/exam.yaml \
+  tracetest test run -d tests/transaction.yaml
+```
+
+4. Follow the link to find the results
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:162]:
```suggestion
4. Follow the link to find the results.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 20]
```diff
@@ -0,0 +1,166 @@
...
+
+## Learn more
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:164]:
```suggestion
## Learn More
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md [hunk 21]
```diff
@@ -0,0 +1,166 @@
...
+
+Feel free to check out our [examples in GitHub](https://github.com/kubeshop/tracetest/tree/main/examples), and join our [Discord Community](https://discord.gg/8MtcMrQNbX) for more info!
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-step-functions-terraform.md:166]:
```suggestion
Please visit our [examples in GitHub](https://github.com/kubeshop/tracetest/tree/main/examples), and join our [Discord Community](https://discord.gg/8MtcMrQNbX) for more info!
```

**Review by jfermi — COMMENTED** [review]
Some editing suggestions.

**Pull request merged**